docs/clusters/how-to-set-up-codec-server.md

```diff
@@ -0,0 +1,46 @@
+---
+id: how-to-set-up-codec-server
+title: How to Set Up a Codec Server?
+sidebar_label: Codec Server
+description: Implementation of a remote encryption/decryption server.
+---
+
+A [Codec Server](/docs/concepts/what-is-a-codec-server) is an optional feature that can further encode or decode Payloads. This can be done through the [Web UI](/docs/web-ui/how-to-use-a-list-filter-in-the-temporal-web-ui) or with [tctl](/docs/tctl/index).
+
+A codec HTTP protocol specifies two endpoints to handle Payload encoding and decoding. Implementations must:
+
+1. Send and receive Payloads protobuf as JSON.
+2. Only check the final part of the incoming URL to determine if the request is for /encode or /decode.
+
+Each platform has a different approach to setting up and running the Codec Server. Follow the steps below to get a Codec Server running on your preferred platform.
+
+## tctl
+
+Before running a Codec Server, make sure that a Temporal service is running.
+
+Start the codec server:
+
+```bash
+go run ./codec-server
+```
+
+Pass the encoded Payloads to the codec endpoint:
+
+```bash
+tctl --codec_endpoint 'http://localhost:{PORT}/{namespace}' workflow show --wid codecserver_workflowID
```

Comment:
Looks as though we need to add `--codec_endpoint` as a [global modifier](https://docs.temporal.io/tctl/#global-modifiers) in the tctl reference. And if you already have that in the works, more power to you.

Reply:
I'll keep this in mind. Thanks!